### DISCUSSION TITLE
How do I make the camera rotate with the object?

### DISCUSSION THREAD

**Discussion by ArtyIF** [ramokz/phantom-camera#442] [Question / Help Wanted]
I've used Cinemachine for my racing game's cameras back when I used Unity, and there the default follow mode rotated with the object it was following. So, if you set the offset to (0, 1, -3), it would always be 3 meters behind (the Z axis is inverted in Unity compared to Godot) the object *along its Z axis* and 1 meter above the object *along its Y axis*.

However, when I tried to do the same in Godot's Phantom Camera, it didn't. Here's an example:

https://github.com/user-attachments/assets/088f23d2-c08d-4b28-8473-64b9a2727e70

Is there any way to make sure the camera follows the object with an offset in its local axis and not global? Only way I can think of is creating a node with a script that does all the following for you, but that defeats a lot of the addon's point for my use case (i.e. to move away from my messy scripting I currently have for my camera and use a user-friendly interface to configure said camera). Maybe the addon's just broken on Godot 4.4-dev7 (yes, I use a dev version for my project, it's just got more stuff for me that I need)?

**Comment by ArtyIF** [comment]
By the way, I did the thing with writing a custom script for the follow point, and realized that the follow damping is also calculated in the global axis.

**Comment by ramokz** [comment]
When you say “rotate with the object”, do you mean making the `PCam` copy the rotational value of the object, i.e., when the object turns the `PCam` rotates by the same mount, or having the `PCam` orbit around the object?

**Reply by ArtyIF** [reply]
Orbit around it, I guess. When the object turns, I want the camera to reposition itself to be behind it.

**Reply by ArtyIF** [reply]
Here's an example with my current custom-made camera system of what I want it to do. PCam seems to just stay in one spot instead.

https://github.com/user-attachments/assets/6c77121f-4226-4706-b104-ada7bc37cafd

**Reply by ramokz** [reply]
There is not an automatic way to do it, but you can use the `Third Person Follow` (https://phantom-camera.dev/follow-modes/third-person) to adjust the camera to rotate around its target.

It would just be a case of calling the rotation function with the same rotation amount you're applying to the followed object; so the object and camera stay in sync.
